core/src/test/scala/com/raphtory/BaseRaphtoryAlgoTest.scala

```diff
@@ -1,61 +1,64 @@
 import com.raphtory.deploy.Raphtory
-import com.raphtory.client.RaphtoryGraph
-import com.raphtory.serialisers.PulsarKryoSerialiser
-import com.typesafe.config.Config
 import com.typesafe.config.Config
-import com.typesafe.config.ConfigFactory
-import com.typesafe.config.ConfigValueFactory
 import com.typesafe.scalalogging.Logger
+import org.apache.commons.io.FileUtils
 import org.apache.pulsar.client.api.Consumer
 import org.apache.pulsar.client.api.Message
-import org.apache.pulsar.client.api.Schema
-import org.scalactic.source
 import org.scalatest.BeforeAndAfter
 import org.scalatest.BeforeAndAfterAll
 import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
 import org.slf4j.LoggerFactory
 
+import java.io.File
+import java.nio.charset.StandardCharsets
 import scala.language.postfixOps
 import scala.reflect.ClassTag
-import scala.util.Random
 import scala.reflect.runtime.universe._
 
-abstract class BaseRaphtoryAlgoTest[T: ClassTag: TypeTag]
+abstract class BaseRaphtoryAlgoTest[T: ClassTag: TypeTag](deleteResultAfterFinish: Boolean = true)
         extends AnyFunSuite
-        with BeforeAndAfterAll {
-  val logger: Logger = Logger(LoggerFactory.getLogger(this.getClass))
-
-  setup()
+        with BeforeAndAfter
+        with BeforeAndAfterAll
+        with Matchers {
 
-  Thread.sleep(5000)
+  val outputDirectory: String = "/tmp/raphtoryTest"
+  val logger: Logger          = Logger(LoggerFactory.getLogger(this.getClass))
 
-  val spout        = setSpout()
-  val graphBuilder = setGraphBuilder()
+  val spout: Spout[T]               = setSpout()
+  val graphBuilder: GraphBuilder[T] = setGraphBuilder()
 
-  val graph            =
-    if (batchLoading) Raphtory.batchLoadGraph[T](spout, graphBuilder)
+  val graph =
+    if (batchLoading()) Raphtory.batchLoadGraph[T](spout, graphBuilder)
     else Raphtory.streamGraph[T](spout, graphBuilder)
-  val conf             = graph.getConfig()
-  val pulsarController = new PulsarController(conf)
 
-  val pulsarAddress: String =
-    conf.getString("raphtory.pulsar.broker.address") //conf.getString("Raphtory.pulsarAddress")
-  val deploymentID: String = conf.getString("raphtory.deploy.id")
+  val conf: Config          = graph.getConfig()
+  val pulsarAddress: String = conf.getString("raphtory.pulsar.broker.address")
+  val deploymentID: String  = conf.getString("raphtory.deploy.id")
+  val pulsarController      = new PulsarController(conf)
 
-  private val kryo: PulsarKryoSerialiser   = PulsarKryoSerialiser()
-  implicit val schema: Schema[Array[Byte]] = Schema.BYTES
-  val testDir                              = "/tmp/raphtoryTest" //TODO CHANGE TO USER PARAM
+  before {
+    setup()
+
+    Raphtory.createClient(
+            deploymentID = "deployment123",
+            customConfig = Map(("raphtory.pulsar.endpoint", "localhost:1234"))
+    )
+  }
+
+  after {
+    if (deleteResultAfterFinish)
+      FileUtils
+        .cleanDirectory(new File(outputDirectory))
+  }
```

Comment:
Is there a way to make the cleanup only run for successful tests? This would require making the cleanup more specific as we only want to delete the folder associated with the current test, not the whole output directory). Would be good to be able to inspect the output of failed tests...